session.go

```diff
@@ -76,16 +77,23 @@ var queryPool = &sync.Pool{
 }
 
 func addrsToHosts(addrs []string, defaultPort int) ([]*HostInfo, error) {
-	hosts := make([]*HostInfo, len(addrs))
-	for i, hostport := range addrs {
+	var hosts []*HostInfo
+	for _, hostport := range addrs {
 		host, err := hostInfo(hostport, defaultPort)
 		if err != nil {
+			// Try other hosts if unable to resolve DNS name
+			if _, ok := err.(*net.DNSError); ok {
+				Logger.Printf("gocql: %v\n", err)
```

Comment:
Thanks for the feedback!

### Given
```go
	cluster := gocql.NewCluster("blahasdf.com", "blahasdf2.com")
	_, err := cluster.CreateSession()
	if err != nil {
		fmt.Printf("err: %s\n", err)
		os.Exit(1)
	}
	os.Exit(0)
```
### Output
```bash
$ go run main.go
2017/04/07 07:14:56 gocql: lookup blahasdf.com: no such host
2017/04/07 07:14:56 gocql: lookup blahasdf2.com: no such host
err: gocql: unable to create session: gocql: failed to resolve any of the provided hostnames
exit status 1
```
I'll add `dns error` to the message to make it more verbose.